src/jinja2/environment.py

```diff
@@ -323,6 +371,14 @@ def __init__(
         self.optimized = optimized
         self.finalize = finalize
         self.autoescape = autoescape
+        if isclass(default_escape):
+            self.default_markup_class = default_escape
+        elif default_escape != html_escape:
+            self.default_markup_class = get_wrapped_escape_class(default_escape)
+        elif callable(self.autoescape) and callable(self.autoescape(None)):
+            self.default_markup_class = get_wrapped_escape_class(self.autoescape(None))
```

Comment:
We call `self.autoescape` twice here, but since we can't use `:=` yet I don't think there's a pretty way of avoiding it...